src/Core/src/AbstractApi.php

```diff
@@ -162,15 +161,4 @@ private function getSigner()
 
         return $this->signer;
     }
-
-    private function fillEndpoint(?string $endpoint): string
```

Comment:
I think this was here for a reason.. 
Ie S3 could have a simpler `getEndpoint` if it didnt needed to bother about the region..

Reply:
Take your time to review, no hurry. I thougth it was here because of the generic AwsClient

Because it's currently not used, I prefer, removing it and re-adding it later when the use case is clear.